src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -65,7 +65,19 @@ export function getEventTagSearchUrl(
   location: Location
 ) {
   const query = {...location.query};
-  query.query = appendTagCondition(query.query, tagKey, tagValue);
+
+  // some tags are dedicated query strings since other parts of the app consumes this,
+  // for example, the global selection header.
+  switch (tagKey.toLowerCase()) {
+    case 'environment':
+      query.environment = tagValue;
+      break;
+    case 'project':
+      query.project = tagValue;
+      break;
+    default:
+      query.query = appendTagCondition(query.query, tagKey, tagValue);
+  }
```

Comment:
@dashed This needs some tests.

Also trying out this multi-line comment feature.